src/initialize-node.ts

```diff
@@ -18,9 +18,10 @@ import { PuppeteerNode } from './node/Puppeteer.js';
 import { PUPPETEER_REVISIONS } from './revisions.js';
 import pkgDir from 'pkg-dir';
 import { Product } from './common/Product.js';
+import { puppeteerDirname } from './compat.js';
 
 export const initializePuppeteerNode = (packageName: string): PuppeteerNode => {
-  const puppeteerRootDirectory = pkgDir.sync(__dirname);
+  const puppeteerRootDirectory = pkgDir.sync(puppeteerDirname);
```

Comment:
pkgDir.sync resolves to the ESM folder now. I think we need to go one level up in case of ESM because of the extra package.json that we add.